tests/nix_tests/test_str_v2.py

```diff
@@ -0,0 +1,140 @@
+    "ibc/C4CFF46FD6DE35CA4CF4CE031E643C8FDC9BA4B99AE598E9B0ED98FE3A2319F9"
+)
+# The ERC20 address of ATOM on Evmos
+ATOM_1_ERC20_ADDRESS = Web3.toChecksumAddress(
+    "0xf36e4C1F926001CEaDa9cA97ea622B25f41e5eB2"
+)
+
+
+@pytest.fixture(scope="module", params=["evmos"])
+def ibc(request, tmp_path_factory):
+    """Prepare the network"""
+    name = "str-v2"
+    evmos_build = request.param
+    path = tmp_path_factory.mktemp(name)
+    # specify the custom_scenario
+    network = prepare_network(path, name, [evmos_build, "cosmoshub-1", "cosmoshub-2"])
+    yield from network
+
+
+def test_str_v2_single_hop(ibc):
+    """
+    Test Single Token Representation v2 with single hop Coin.
+    It should automatically create an ERC20 precompiled contract.
+    And register a token pair.
+    """
+    assert_ready(ibc)
+
+    evmos: Evmos = ibc.chains["evmos"]
+    gaia: CosmosChain = ibc.chains["cosmoshub-1"]
+
+    evmos_cli = evmos.cosmos_cli()
+    evmos_addr = ADDRS["signer2"]
+    bech_dst = eth_to_bech32(evmos_addr)
+
+    gaia_cli = gaia.cosmos_cli()
+    gaia_addr = gaia_cli.address("signer2")
+
+    old_dst_balance = get_balance(evmos, bech_dst, ATOM_IBC_DENOM)
+    rsp = gaia_cli.ibc_transfer(
+        gaia_addr, bech_dst, "5000uatom", "channel-0", 1, fees="10000uatom"
+    )
+    assert rsp["code"] == 0
+
+    # NOTE: Sleep some time because wait_for_fn doesn't work for some reason ?
+    time.sleep(30)
+
+    w3 = evmos.w3
+    active_dynamic_precompiles = evmos_cli.evm_params()["params"][
+        "active_dynamic_precompiles"
+    ]
+    new_dest_balance = get_balance(evmos, bech_dst, ATOM_IBC_DENOM)
+    erc_dest_balance = erc20_balance(w3, ATOM_1_ERC20_ADDRESS, evmos_addr)
+
+    assert len(active_dynamic_precompiles) == 1
+    assert old_dst_balance + 5000 == new_dest_balance
+    assert old_dst_balance + 5000 == erc_dest_balance
+
+
+def test_str_v2_multi_hop(ibc):
+    """
+    Test Single Token Representation v2 with multi hop Coin.
+    It should NOT create an ERC20 precompiled contract and token pair.
+    """
+    assert_ready(ibc)
+
+    evmos: Evmos = ibc.chains["evmos"]
+    gaia: CosmosChain = ibc.chains["cosmoshub-1"]
+    gaia2: CosmosChain = ibc.chains["cosmoshub-2"]
+
+    evmos_cli = evmos.cosmos_cli()
+    evmos_addr = ADDRS["signer2"]
+    bech_dst = eth_to_bech32(evmos_addr)
+
+    # The starting balance of the destination address
+    evmos_old_balance = get_balance(evmos, bech_dst, ATOM_2_IBC_DENOM_MULTI_HOP)
+
+    # Cosmos hub 1
+    gaia_cli = gaia.cosmos_cli()
+    gaia_addr = gaia_cli.address("signer2")
+    gaia1_old_balance = get_balance(gaia, gaia_addr, ATOM_1_IBC_DENOM_ATOM_2)
+
+    # Cosmos hub 2
+    gaia2_cli = gaia2.cosmos_cli()
+
+    rsp = gaia2_cli.ibc_transfer(
+        gaia_addr, gaia_addr, "50000uatom", "channel-1", 1, fees="10000uatom"
+    )
+    assert rsp["code"] == 0
+
+    # NOTE: Using time.sleep here because wait_for_fn doesn't work for some reason
+    time.sleep(30)
```

Comment:
same question here

Reply:
Why is this needed ? We should use the `wait_for_fn` function here